docs/Running-Mastodon/Production-guide.md

```diff
@@ -1,99 +1,150 @@
+    curl -sL https://deb.nodesource.com/setup_4.x | sudo bash -
 
-  ssl_certificate     /etc/letsencrypt/live/example.com/fullchain.pem;
-  ssl_certificate_key /etc/letsencrypt/live/example.com/privkey.pem;
+    sudo apt-get install nodejs
 
-  keepalive_timeout    70;
-  sendfile             on;
-  client_max_body_size 0;
-  gzip off;
+    sudo npm install -g yarn
 
-  root /home/mastodon/live/public;
+## Nginx w/ HTTPS via LetsEncrypt
+First, make sure your domain name is either pointing to your server's IP via A record, or to its hostname via CNAME record. Or, if you have a hosting setup where you can just set the DNS servers entirely, use that. Regardless, your DNS has to be configured before LetsEncrypt will work.
 
-  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains";
+### LetsEncrypt (SSL is free and easy, do this!)
+Go to https://certbot.eff.org/ to get setup instructions for your server configuration. If you're following this guide, you want the nginx setup - and for the purposes of this example, let's assume you're running on Ubuntu 16.04. (If you're not the instructions may slightly vary - follow what it says on the website instead of what it says here).
 
-  location / {
-    try_files $uri @proxy;
-  }
+Certbot tells us to:
 
-  location @proxy {
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto https;
+    sudo add-apt-repository ppa:certbot/certbot
+    sudo apt-get update
+    sudo apt-get install certbot 
 
-    proxy_pass_header Server;
+Depending on your Ubuntu image, you may first need to run: 
 
-    proxy_pass http://localhost:3000;
-    proxy_buffering off;
-    proxy_redirect off;
-    proxy_http_version 1.1;
-    proxy_set_header Upgrade $http_upgrade;
-    proxy_set_header Connection $connection_upgrade;
+    sudo apt-get install software-properties-common python-software-properties
 
-    tcp_nodelay on;
-  }
+Once that's installed, run:
 
-  location /api/v1/streaming {
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto https;
+    certbot certonly
 
-    proxy_pass http://localhost:4000;
-    proxy_buffering off;
-    proxy_redirect off;
-    proxy_http_version 1.1;
-    proxy_set_header Upgrade $http_upgrade;
-    proxy_set_header Connection $connection_upgrade;
+and follow the prompts. It'll ask you what domain name you want to get a certificate for - put in the public facing domain name, which should be the same as your instance name. The one you set up DNS for. This installer is cool, it has the ability to stand up a temporary webserver on your host that it will use to verify that you own the domain name you're claiming. Once this is done, great! You can provide HTTPS access!
```

Comment:
There's one problem with using the standalone challenge here: Renewal won't work. That's because `nginx` will occupy port 80 and 443 once you're done with this guide. Probably the best option for this use-case is the webroot challenge. You can point it to nginx's default value for the `root` directive (probably `/var/www/html`?) and later change it to mastodon's public directory in the corresponding renewal file in `/etc/letsencrypt/renewal`.